### PR TITLE
add doc for [attribute] in ALIGN, OFFSET, and POSITION

### PR DESCRIPTION
**Pull request by danduk82**
This is documentation relative to the [following PR](https://github.com/mapserver/mapserver/pull/6052) in mapserver repository.

I noticed that the documentation for `[attribute]` in POSITION was missing, so I added it too.

### CODE REVIEW

File: en/mapfile/label.txt [hunk 1]
```diff
@@ -10,12 +10,30 @@
...
     Specifies text alignment for multiline labels (see WRAP_) Note
     that the alignment algorithm is far from precise, so don't expect
     fabulous results (especially for *right* alignment) if you're not
     using a fixed width font.
 
+
+    [attribute] was added in version 7.6. If an attribute is used,
+    it is expected to be:
+
+    *  of type *string* and should be one of the following: *left*, *center*, *right*.
```
Comment (havatv) [inline en/mapfile/label.txt:23]:
```suggestion
    * of type *string* and should be one of the following: *left*, *center*, *right*.

```

File: en/mapfile/label.txt [hunk 2]
```diff
@@ -10,12 +10,30 @@
...
+    * of type *integer* and should be [1,2,3] which means:
+
+      * 1 = "left"
+      * 2 = "center"
+      * 3 = "right"
+
+
+    if the [attribute] is of type *integer* as described above, the label
+    generation will be much faster as no string comparison are made.
+
```
Comment (havatv) [inline en/mapfile/label.txt:33]:
```suggestion
```

File: en/mapfile/label.txt [hunk 3]
```diff
@@ -10,12 +10,30 @@
...
+
```
Comment (havatv) [inline en/mapfile/label.txt:34]:
```suggestion
```

File: en/mapfile/label.txt [hunk 4]
```diff
@@ -10,12 +10,30 @@
...
+
```
Comment (havatv) [inline en/mapfile/label.txt:35]:
```suggestion
```

File: en/mapfile/label.txt [hunk 5]
```diff
@@ -10,12 +10,30 @@
...
+    [attribute] was added in version 7.6. If an attribute is used,
+    it is expected to be:
+
+    *  of type *string* and should be one of the following: *left*, *center*, *right*.
+    * of type *integer* and should be [1,2,3] which means:
+
+      * 1 = "left"
+      * 2 = "center"
+      * 3 = "right"
+
```
Comment (havatv) [inline en/mapfile/label.txt:29]:
```suggestion
```

File: en/mapfile/label.txt [hunk 6]
```diff
@@ -10,12 +10,30 @@
...
    pair: LABEL; ALIGN
 
-ALIGN [left|center|right]
+ALIGN [left|center|right|attribute]
     Specifies text alignment for multiline labels (see WRAP_) Note
     that the alignment algorithm is far from precise, so don't expect
     fabulous results (especially for *right* alignment) if you're not
     using a fixed width font.
 
+
```
Comment (havatv) [inline en/mapfile/label.txt:19]:
```suggestion
```

File: en/mapfile/label.txt [hunk 7]
```diff
@@ -287,27 +305,32 @@ MINSCALEDENOM [double]
...
-    Offset values for labels, relative to the lower left hand corner of the 
-    label and the label point. Given in pixels. In the case of rotated text 
-    specify the values as if all labels are horizontal and any rotation will 
+OFFSET [x|attribute_x][y|attribute_y]
+    Offset values for labels, relative to the lower left hand corner of the
+    label and the label point. Given in pixels. In the case of rotated text
+    specify the values as if all labels are horizontal and any rotation will
     be compensated for.
 
+
```
Comment (havatv) [inline en/mapfile/label.txt:329]:
```suggestion
```

File: en/mapfile/label.txt [hunk 8]
```diff
@@ -287,27 +305,32 @@ MINSCALEDENOM [double]
...
+    [attribute_x] and [attribute_y] were added in version 7.6. If an attribute
+    is used, it must be of type *integer*.
+
```
Comment (havatv) [inline en/mapfile/label.txt:332]:
```suggestion
```

**Comment by jmckenna** [comment]
note that this will have to be merged after the source code build errors are fixed & merged through https://github.com/mapserver/mapserver/pull/6052

**Comment by danduk82** [comment]
@havatv I just accepted your suggestions here. Is this OK like this? I guess you will squash the branch before merging. Otherwise I can rebase it a forcefully push it again, but I dislike this approach in general.

**Comment by havatv** [comment]
> @havatv I just accepted your suggestions here. Is this OK like this? I guess you will squash the branch before merging. Otherwise I can rebase it a forcefully push it again, but I dislike this approach in general.

There is an easier way, but it is too well hidden... ;-)
If you go to the "Files changed" tab, you will find an "Add suggestion to batch" button under each suggestion. Use that button on the suggestions you accept, and when you have finished adding suggestions, you can commit them all in one go by clicking the button with the label Commit suggestions. https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

File: en/mapfile/label.txt [hunk 9]
```diff
@@ -10,12 +10,27 @@
...
+    * of type *string* and should be one of the following: *left*, *center*, *right*.
+
+    * of type *integer* and should be [1,2,3] which means:
+
+      * 1 = "left"
+      * 2 = "center"
+      * 3 = "right"
+
+    if the [attribute] is of type *integer* as described above, the label
+    generation will be much faster as no string comparison are made.
```
Comment (asaunier) [inline en/mapfile/label.txt:31]:
```suggestion
    generation will be much faster as no string comparison is made.

```

**Comment by danduk82** [comment]
> > @havatv I just accepted your suggestions here. Is this OK like this? I guess you will squash the branch before merging. Otherwise I can rebase it a forcefully push it again, but I dislike this approach in general.
> 
> It is OK like this. There is a way to combine suggestion into one commit on github.com, but it is too well hidden... ;-)
> If you go to the "Files changed" tab, you will find an "Add suggestion to batch" button under each suggestion. Use that button on the suggestions you accept, and when you have finished adding suggestions, you can commit them all in one go by clicking the button with the label Commit suggestions. https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

Thanks I didn't know.

File: en/mapfile/label.txt [hunk 10]
```diff
@@ -10,12 +10,28 @@
...
     that the alignment algorithm is far from precise, so don't expect
     fabulous results (especially for *right* alignment) if you're not
     using a fixed width font.
 
+    [attribute] was added in version 7.6. If an attribute is used,
+    it is expected to be:
+
+    * of type *string* and should be one of the following: *left*, *center*, *right*.
+
+    * of type *integer* and should be [1,2,3] which means:
```
Comment (yjacolin) [inline en/mapfile/label.txt:24]: not sure this is to be added. This kind of value could work everywhere but doc didn't talk about this anywhere.

**Comment by yjacolin** [comment]
Features have been merged in MapServer source code. We can merge the doc now.

**Comment by jmckenna** [comment]
thank you @danduk82 and @yjacolin !!

**Pull request merged**